eclipse-tractusx/bpdm
# Feat: Add Central-IDP as Default Authentication Server

## Description


Add Central-IDP as default authentication service to BPDM applications

- adapted default application configuration in application properties
- added Central-IDP as dependency to the BPDM charts
- adapted the tests to work with Central-IDP
- exchanged the test realm to match Central-IDP's default realm

### How To test

Local:
1. Remove the outdated BPDM Keycloak docker container (if existing)
2. Setup the Central-IDP Keycloak via docker compose
3. Execute all BPDM applications in default configuration and test the golden record process

Charts:
1. Docker build all BPDM applications and push them into a registry that can be accessed by your Kubernetes environment (see chart testing workflow)
2. Change the BPDM chart values to match your registry
3. Install the BPDM chart and observe whether all applications can communicate with each other

## Note

This pull request has a lot of line changes. However, most of these changes result from exchanging the previous Keycloak realm test file with the Central-IDP's realm. Most of these entries in that realm are not relevant for the execution of BPDM.

This pull request uses the following state of the Central-IDP as contributed in: https://github.com/eclipse-tractusx/portal-iam/issues/154

Fixes #994 



## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files
